routers/api/scouting.ts

```diff
@@ -424,4 +427,53 @@ router.get(
     }
 );
 
+router.get("/leaderboard", requireScoutingAuth, async (ctx, next) => {
+    addAPIHeaders(ctx);
+    try {
+        const timeframe = ctx.query.timeframe || 'all';
+        const dateFilter: any = {};
+        
+        if (timeframe === 'week') {
+            const lastWeek = new Date();
+            lastWeek.setDate(lastWeek.getDate() - 7);
+            dateFilter.serverTimestamp = { $gte: lastWeek.getTime() };
+        } else if (timeframe === 'day') {
+            const yesterday = new Date();
+            yesterday.setDate(yesterday.getDate() - 1);
+            dateFilter.serverTimestamp = { $gte: yesterday.getTime() };
+        }
+
+        const leaders = await ScoutingEntry.aggregate([
+            { $match: dateFilter },
+            {
+                $group: {
+                    _id: {
+                        team: "$contributor.team",
+                        username: "$contributor.username"
+                    },
+                    scansCount: { $sum: 1 },
+                    totalXp: { $sum: "$xp" },
+                    totalNuts: { $sum: "$nuts" },
+                    totalBolts: { $sum: "$bolts" }
+                }
+            },
+            { $sort: { scansCount: -1 } },
+            { $limit: 50 }
+        ]);
+
+        ctx.body = {
+            success: true,
+            body: {
+                leaders
+            }
+        };
+    } catch (error) {
+        console.error('Error fetching leaderboard:', error);
+        ctx.body = {
+            success: false,
+            error: 'Internal server error'
```

Comment:
give a more descriptive error message (like "Unable to fetch leaderboard, please try again later.")